lib/AutoPodArgsPass.cpp

```diff
@@ -141,3 +168,35 @@ void AutoPodArgsPass::AddMetadata(Function &F, clspv::PodArgImpl impl) {
           IntegerType::get(F.getContext(), 32), static_cast<uint32_t>(impl))));
   F.setMetadata(clspv::PodArgsImplMetadataName(), md);
 }
+
+bool AutoPodArgsPass::ContainsArrayType(Type *type) const {
+  if (isa<ArrayType>(type)) {
+    return true;
+  } else if (auto struct_ty = dyn_cast<StructType>(type)) {
+    for (auto sub_type : struct_ty->elements()) {
+      if (ContainsArrayType(sub_type))
+        return true;
+    }
+  }
+
+  return false;
+}
+
+bool AutoPodArgsPass::ContainsSizedType(Type *type, uint32_t width) const {
+  if (auto int_ty = dyn_cast<IntegerType>(type)) {
+    return int_ty->getBitWidth() == width;
+  } else if (type->isHalfTy()) {
+    return true;
```

Comment:
This should be return width == 16.

Reply:
```
isHalfTy () const
--
  | Return true if this is 'half', a 16-bit IEEE fp type
```

That's from LLVM's docs. LLVM doesn't have a derived type for floating point types.

Reply:
oh nvm i see what you meant.